### DISCUSSION TITLE
Mapbox-gl/deck interleaved lighting

### DISCUSSION THREAD

**Discussion by MartinP-C** [visgl/deck.gl#7322] [General]
I am using a GeoJsonLayer with extruded shapes inside of ReactMapGL framework, using the [MapboxLayer](https://deck.gl/docs/api-reference/mapbox/mapbox-layer).

It is working very well!

However now I want to adjust the lighting. Where is the lighting for the Deck GeoJsonLayer coming from?
I have no Deck instance (at least, not one created by me) in order to use LightingEffect.
Changing the Mapbox/ReactMapGl `light` prop doesn't seem to affect the GeoJsonLayer rendering.

Is the light source controlled by Mapbox, or by Deck?
If it is controlled by Deck, how do I adjust it while it is being used inside a Mapbox custom layer?

**Comment by Pessimistress** [comment]
I recommend you use the new [MapboxOverlay](https://deck.gl/docs/api-reference/mapbox/mapbox-overlay) in order to leverage all `Deck` props ([effects](https://deck.gl/docs/developer-guide/using-lighting) in this case).
